src/modals/MediaDbSearchModal.ts

```diff
@@ -99,10 +102,24 @@ export class MediaDbSearchModal extends Modal {
 			const apiToggleComponent = new ToggleComponent(apiToggleComponentWrapper);
 			apiToggleComponent.setTooltip(unCamelCase(mediaType));
 			apiToggleComponent.setValue(this.selectedTypes.find(x => x.name === mediaType).selected);
+			if (apiToggleComponent.getValue()) {
+				currentToggle = apiToggleComponent;
+			}
 			apiToggleComponent.onChange(value => {
-				this.selectedTypes.find(x => x.name === mediaType).selected = value;
+				if (value) {
```

Comment:
What exactly is this code doing? Are you forcing the value of the toggle to be off in certain conditions? If so there is a better way to disable the toggle component. See the `setDisabled(value)` prop of the toggle component.